### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by SanketDhabarde**
### Changes made
- Added login page
- Added signup page

### screenshot login page
Screenshot (196)

### screenshot signup page
Screenshot (197)

### CODE REVIEW

File: src/pages/Auth/Login.jsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  const loginHandler = async (e) => {
+    e.preventDefault();
+    try {
+      const { data } = await axios.post("/api/auth/login", { email, password });
+      const { foundUser, encodedToken } = data;
+      setUser(foundUser);
+      localStorage.setItem("token", encodedToken);
+      navigate("/home");
+    } catch (error) {
+      console.error(error);
```
Comment (anupkgurung) [inline src/pages/Auth/Login.jsx:25]: console can be removed as you have a setter of error

**Review by anupkgurung — APPROVED** [review]
clean code.
good to see astrik for selection in index

File: src/pages/Auth/Login.jsx (hunk 1 above)
Comment (SanketDhabarde) [reply src/pages/Auth/Login.jsx:25]: yes. but console added from dev perspective if anything went wrong we should know the errors

**Review by ianuragsharma — APPROVED** [review]
-Code looks clean and understandable.
-Folder structure is so good.
-Good use of context and reducer.
-Looks good to merge.

**Pull request merged**